app/services/support_interface/change_application_choice_course_option.rb

```diff
@@ -34,6 +35,12 @@ def check_application_state!
       raise "Changing the course option of application choices in the #{application_choice.status} state is not allowed"
     end
 
+    def check_interviewing_providers!
+      return if !application_choice.interviewing? || (application_choice.interviewing? && application_choice.provider_ids.include?(provider_id))
+
+      raise 'Changing a course choice when the provider is not on the interview is not allowed'
```

Comment:
See earlier comment about exception / error types. Raising a `RuntimeError` here means you have to catch something quite generic and interpret it as something quite specific which allows for a different/unexpected `RuntimeError` to be considered as the same sort of problem.